packages/block-library/src/blocks/slideshow/init.php

```diff
@@ -158,3 +125,112 @@ class="<?php echo esc_attr( join( ' ', $classes ) ); ?>"
 		return ob_get_clean();
 	}
 }
+
+function novablocks_get_post_slide_markup( $post, $attributes ) {
+
+	$slider_background_src = get_the_post_thumbnail_url( $post );
+	$thumbnail = wp_get_attachment_metadata( get_post_thumbnail_id($post) );
+
+	$width = $thumbnail['sizes']['large']['width'];
```

Comment:
There is the change for the attachment not having the large image size.

Markup output should be abstracted into a function that receives $image or $image_id and $content and outputs them with the proper wrappers:
```php
	<div class="<?php echo esc_attr( $className ); ?>">
		<div class="novablocks-slideshow__slide-wrap">
			<div class="novablocks-slideshow__background novablocks-u-background">
				<div class="novablocks-mask">
					<div class="novablocks-parallax">
					    <?php novablocks_output_slideshow_slide_background( $image or $image_id ); ?>
					....
			<div class="novablocks-slideshow__foreground novablocks-foreground novablocks-u-content-padding novablocks-u-content-align">
				<div class="novablocks-slideshow__inner-container novablocks-u-content-width">
				<?php echo $content; ?>
				...
```
in order to not repeat yourself and have consistent markup for both use cases